File: java/ecbenchmark/src/ecbenchmark/operator/Rotated.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.operator;

import ecbenchmark.Function;
import ecbenchmark.Operator;
import ecbenchmark.util.CommonMath;

/**
 *
 * @author jcrada
 */
public class Rotated extends Operator{

    protected double[][] rotation;

    public Rotated(double[][] rotation, Function f) {
        super(f);
        this.rotation = rotation;
    }

    @Override
    public double f(final double[] x) {
        double[] rotated_x = CommonMath.Multiply(x, rotation);
        return innerFunction.f(rotated_x);
    }
}

File: cpp/ecbenchmark/src/operator/Rotated.cpp
#include "ecbenchmark/operator/Rotated.h"
#include "ecbenchmark/util/CommonMath.h"

namespace ecb {

    Rotated::Rotated(const std::vector<std::vector<scalar> >& rotation, Function* f)
    : Operator(f), _rotation(rotation) {

    }

    void Rotated::setRotation(const std::vector<std::vector<scalar> >& rotation) {
        this->_rotation = rotation;
    }

    std::vector<std::vector<scalar> > Rotated::rotation() const {
        return this->_rotation;
    }

    scalar Rotated::f(const std::vector<scalar>& x) {
        std::vector<scalar> rotated_x = CommonMath::Multiply(x, _rotation);
        return _innerFunction->f(rotated_x);
    }
}
